Academics crud

renangabriel27 opened the pull request:
- rails db:populate (to generate academics data)

utfpr-gp-tsi commented:
Por que foi mudado de professor para admin? Pelo que lembro o admin será um professor, seria o professor responsável, certo?

renangabriel27 commented:
> Por que foi mudado de professor para admin? Pelo que lembro o admin será um professor, seria o professor responsável, certo?

Corrigido

utfpr-gp-tsi commented on app/views/professors/academics/_form.html.erb:
```diff
@@ -0,0 +1,20 @@
+<%= simple_form_for [:professors, @academic] do |f| %>
+  <%= f.error_notification %>
+  <%= f.error_notification message: f.object.errors[:base].to_sentence if f.object.errors[:base].present? %>
```
The lines 2 and 3 can be removed. This way it shows two messages: 

![image](https://user-images.githubusercontent.com/22449115/52181618-e2076500-27da-11e9-9cfa-9015c9fafb39.png)


It's necessary just one, that can be: "Alguns erros foram encontrados, por favor verifique:"

utfpr-gp-tsi commented on app/models/academic.rb:
```diff
@@ -0,0 +1,17 @@
+  enum gender: { male: 'M', female: 'F' }, _prefix: :gender
+
+  validates :name,   presence: true
+  validates :ra,     presence: true
```
The RA must be unique.

utfpr-gp-tsi commented on app/controllers/professors/academics_controller.rb:
```diff
@@ -0,0 +1,71 @@
+                 only: [:edit]
+
+  def index
+    @academics = Academic.all
```
The order mas be ASC by name. Take care about accents.

utfpr-gp-tsi commented on spec/features/professors/professors_update_spec.rb:
```diff
@@ -0,0 +1,46 @@
+  end
+
+  it 'should not update a professor with invalid fields' do
+    fill_in Professor.human_attribute_name(:email), with: 'email'
```
It's preferable to use the ID, as you did in other tests.

renangabriel27 commented:
> I guess the namespace to register professors and academics must be /responsible. We can talk more on Tuesday about the organization. But It more easy to manage with roles and roles give access to the namespace.

fixed

renangabriel27 commented:
> * Runing rubocop it find 72 files inspected, 37 offenses detected.
> * Its necessary a pagination, um can use kaminari.
> * Its necessary a search, take care about accents.

fixed

renangabriel27 commented:
> 1. Show 25 academics by page.
> 2. When it appears "100 registros" should be "apresentando 25 de 100 registros"
> 3. When you use the pagination the url doesn't change, this cause a problem. Image that I'm seeing page 10, and click to see a academic details, when I click back, it's back to the first page, not to the page that I have. The same happened use the search, or  combination of search and pagination.
> 4. When I updated a academic, the message "Acadêmico Atualizado com sucesso!" is presented, but when a navigate by pagination the message doesn't disappear. You can put a fade-out in 10 seconds.
> 5. The populate isn't idented. You must consider it to rubocop checks.

fixed

utfpr-gp-tsi commented on app/views/responsible/academics/_academic.html.erb:
```diff
@@ -0,0 +1,22 @@
+<% @academics.each do |academic| %>
```
This will show the same register more than once.

utfpr-gp-tsi commented on app/views/responsible/academics/index.html.erb:
```diff
@@ -0,0 +1,35 @@
+    <% end %>
+  </div>
+</div>
+
```
Show this information:
```
<%= page_entries_info @posts %>
#=> Displaying items 6 - 10 of 26 in total
```

utfpr-gp-tsi commented on app/views/responsible/academics/index.html.erb:
```diff
@@ -0,0 +1,35 @@
+  </table>
+</div>
+
+<%= paginate @academics %>
```
![image](https://user-images.githubusercontent.com/22449115/53341789-3e960580-38eb-11e9-9633-236b8e02137d.png)

The page number doesn't change properly.
